### PR TITLE
Thread safety

### PR DESCRIPTION
**Pull request by RyanS-inin**
These were all the changes needed to make ApiClient thread-safe, as best as I can tell.  There are still a few gotchas (such as one thread being able to reconfigure a client that a second thread is using), but it should at least be thread-safe as long as the client is careful.

### CODE REVIEW

File: swagger_template/ApiClient.mustache [hunk 1]
```diff
@@ -619,9 +623,12 @@ public class ApiClient {
...
             }
 
-            statusCode = response.getHttpResponse().getStatusLine().getStatusCode();
-            responseHeaders = response.getHttpResponse().getAllHeaders();
-
+            int statusCode = response.getHttpResponse().getStatusLine().getStatusCode();
+            Header[] responseHeaders = response.getHttpResponse().getAllHeaders();
+            
+            this.statusCode = statusCode;
+            this.responseHeaders = responseHeaders;
```
Comment (RyanS-inin) [inline swagger_template/ApiClient.mustache]: I left instance variables in to keep backward-compatability, but I'm not sure how useful they actually are.

File: swagger_template/ApiClient.mustache [hunk 2]
```diff
@@ -323,7 +323,9 @@ public class ApiClient {
      */
     public Date parseDate(String str) {
         try {
-            return dateFormat.parse(str);
+            synchronized(dateFormat) {
+                return dateFormat.parse(str);
```
Comment (T-Boss) [inline swagger_template/ApiClient.mustache]: What is the purpose of synchronizing this? Does dateFormat.parse() conflate the results if multiple calls are made simultaneously?

File: swagger_template/ApiClient.mustache (hunk 1 above)
Comment (T-Boss) [inline swagger_template/ApiClient.mustache]: I'll remove this kind of stuff when I add async support. This pattern doesn't make a lot of sense in the first place. It came from the main swagger-codegen repo, so it wasn't an intentional feature on my part.

File: swagger_template/ApiClient.mustache (hunk 2 above)
Comment (T-Boss) [inline swagger_template/ApiClient.mustache]: Per conversation with Ryan, the javadocs state that this is not threadsafe in Java 7.

**Pull request merged**